xls/dslx/bytecode/bytecode_interpreter_test.cc

```diff
@@ -1253,6 +1253,68 @@ fn main() -> MyU9 {
   EXPECT_THAT(value.GetBitValueUint64(), IsOkAndHolds(0x1ff));
 }
 
+TEST(BytecodeInterpreterTest, AssertEqFailProcs) {
+  constexpr std::string_view kProgram = R"(
+#[test_proc]
+proc BTester {
+    terminator: chan<bool> out;
+
+    init { (u32:0) }
+
+    config(terminator: chan<bool> out) {
+        (terminator,)
+    }
+
+    next(tok: token, state: u32) {
+        let count = state;
+        let _ = assert_eq(count, u32:0);
+        let tok = send_if(tok, terminator, count > u32:1, true);
```

Comment:
why do we need the `send_if` here?

Reply:
It's necessary, as I found that first evaluation will yield correct error message, that is why I have to delay send, so that in first iteration send will not happen, and in second iteration assert will fail

Reply:
ah, I see maybe you can add a comment to make that clearer? also since we are only testing dslx interpreter here (and not ir conversion) maybe concurrently send would work?

Reply:
I tried it. Error only triggers after new interpreter frame is created, so test has to reach end of next function at least once and start once again.